modal/_asgi.py

```diff
@@ -204,45 +204,54 @@ async def request_generator():
                 if not message.get("more_body", False):
                     break
             elif message["type"] == "http.disconnect":
-                proxy_response.connection.transport.abort()  # Abort the connection.
-                break
+                if proxy_response:
+                    proxy_response.connection.transport.abort()  # Abort the connection.
+                else:
+                    raise NameError("Proxy response not defined")
```

Comment:
Is it possible for proxy_response to exist while the request generator is still running? If not, we could remove the `if proxy_response` case.

`NameError` works but could be better represented as [`ConnectionAbortedError`](https://docs.python.org/3/library/exceptions.html#ConnectionAbortedError) since this is caused by an aborted request.